allennlp/data/fields/text_field.py

```diff
@@ -14,50 +14,51 @@
     to tokenize raw strings using a :class:`..tokenizers.Tokenizer`.
 
     Because string tokens can be represented as indexed arrays in a number of ways, we also take a
-    list of :class:`TokenIndexer` objects that will be used to convert the tokens into indices.
+    dictionary of :class:`TokenIndexer` objects that will be used to convert the tokens into indices.
     Each ``TokenIndexer`` could represent each token as a single ID, or a list of character IDs, or
     something else.
 
-    This field will get converted into a list of arrays, one for each ``TokenIndexer``.  A
+    This field will get converted into apy dictionary of arrays, one for each ``TokenIndexer``.  A
     ``SingleIdTokenIndexer`` produces an array of shape (num_tokens,), while a
     ``TokenCharactersIndexer`` produces an array of shape (num_tokens, num_characters).
     """
-    def __init__(self, tokens: List[str], token_indexers: List[TokenIndexer]) -> None:
+    def __init__(self, tokens: List[str], token_indexers: Dict[str, TokenIndexer]) -> None:
         self._tokens = tokens
         self._token_indexers = token_indexers
-        self._indexed_tokens = None  # type: Optional[List[TokenList]]
+        self._indexed_tokens = None  # type: Optional[Dict[str, TokenList]]
 
-    @overrides
+    # @overrides
     def count_vocab_items(self, counter: Dict[str, Dict[str, int]]):
-        for indexer in self._token_indexers:
+        for indexer in self._token_indexers.values():
             for token in self._tokens:
                 indexer.count_vocab_items(token, counter)
 
-    @overrides
+    # @overrides
     def index(self, vocab: Vocabulary):
-        token_arrays = []
-        for indexer in self._token_indexers:
+        token_arrays = {}
+        for indexer_name, indexer in self._token_indexers.items():
             arrays = [indexer.token_to_indices(token, vocab) for token in self._tokens]
-            token_arrays.append(arrays)
+            token_arrays[indexer_name] = arrays
         self._indexed_tokens = token_arrays
 
-    @overrides
+    # @overrides
     def get_padding_lengths(self) -> Dict[str, int]:
         lengths = []
         if self._indexed_tokens is None:
             raise ConfigurationError("You must call .index(vocabulary) on a "
                                      "field before determining padding lengths.")
-        for indexer, array in zip(self._token_indexers, self._indexed_tokens):
+        for indexer_name, indexer in self._token_indexers.items():
             indexer_lengths = {}
 
             # This is a list of dicts, one for each token in the field.
-            token_lengths = [indexer.get_padding_lengths(token) for token in array]
+            token_lengths = [indexer.get_padding_lengths(token) for token in self._indexed_tokens[indexer_name]]
             # TODO(Mark): This breaks if the token list is empty, but we need to be able to have empty fields.
```

Comment:
No, the problem occurs when we pad a `ListField` to a maximum length with empty `TextFields`, because we then call get padding lengths with a `TextField` containing no tokens, but we expect below (`for key in token_lengths[0].keys()`) that there is a minimum of 1 element in `token_fields`. I left this previously but I think I can fix it now.